### ISSUE TITLE
Implement Database Schema for Assignment Management

### ISSUE THREAD

**Issue by jlgsl** [MatthewFuchs/Ogopogo-Tech#87]
## Overview
Before we can proceed with the full implementation of the Assignment Management feature on the frontend, we need to establish the underlying database schema to store and manage the assignment data effectively.

## Tasks
- [x] Define the assignment schema including necessary fields such as `title`, `description`, `dueDate`, and `teacherId`.
- [x] Set up the MongoDB collection for assignments within our current database environment.
- [x] Ensure the schema supports references or embedded documents for future features like questions and submissions.
- [x] Implement model validations to ensure data integrity when assignments are created or updated.
- [x] Develop basic CRUD operations for assignments in the backend, including API endpoints for creating, reading, updating, and deleting assignments.

## Acceptance Criteria
- The `assignments` collection is successfully created in the MongoDB database.
- Assignment documents can be added with validation rules ensuring data integrity.
- The backend provides API endpoints to perform CRUD operations on the `assignments` collection.
- API endpoints are tested and verified to handle requests and responses appropriately.

## Dependencies
- Completion of the user authentication system to manage `teacherId` references.

## Out of Scope
- Frontend implementation of the Assignment Management page.
- Automatic grading features.
- Integration with the questions or submissions collections.

This task will serve as the foundation for further assignment-related features and functionality. Let's aim to keep the schema flexible to accommodate any foreseeable changes as we get feedback from the implementation of related features.
